lib/Dialect/Arc/ArcCostModel.cpp

```diff
@@ -16,52 +16,66 @@ using namespace circt;
 using namespace arc;
 using namespace std;
 
-size_t ArcCostModel::getCost(Operation *op) { return computeOperationCost(op); }
+// FIXME: May be refined and we have more accurate operation costs
+enum class OperationCost : size_t {
+  NOCOST,
+  NORMALCOST,
+  PACKCOST = 2,
+  EXTRACTCOST = 3,
+  CONCATCOST = 3,
+  SAMEVECTORNOSHUFFLE = 0,
+  SAMEVECTORSHUFFLECOST = 2,
+  DIFFERENTVECTORNOSHUFFLE = 2,
+  DIFFERENTVECTORSHUFFLECOST = 3
+};
 
-size_t ArcCostModel::computeOperationCost(Operation *op) {
+OperationCosts ArcCostModel::getCost(Operation *op) {
+  return computeOperationCost(op);
+}
+
+OperationCosts ArcCostModel::computeOperationCost(Operation *op) {
   if (opCostCash.count(op))
     return opCostCash[op];
```

Comment:
You can often use iterators to combine checking if a value exists in a map, and then doing something with the value if it does exist:
```suggestion
  if (auto it = opCostCache.find(op); it != opCostCache.end())
    return it->second;
```